core/base/src/main/java/org/eclipse/dataspaceconnector/core/CoreServicesExtension.java

```diff
@@ -211,7 +213,10 @@ public OkHttpClient addHttpClient(ServiceExtensionContext context) {
 
     @Provider
     public EventRouter eventRouter(ServiceExtensionContext context) {
-        return new EventRouterImpl(context.getMonitor());
+        var executor = context.hasService(EventExecutorServiceContainer.class) ?
```

Comment:
This should be done with `@Provider(isDefault=true)` and `@Inject(required=false)`. That way, it would contribute the `ExecutorService` back to the `context`

Reply:
You're almost there. Please use `@Inject(required=false)`, so avoid an exception when no `EventExecutorServiceContainer` is registered.

Reply:
I addressed your last comment. Thanks, @paullatzelsperger.